src/docfx/lib/ProcessUtility.cs

```diff
@@ -181,81 +179,30 @@ public static bool ReleaseExclusiveLock(string lockName, string acquirer)
         }
 
         /// <summary>
-        /// Start a new process and wait for its execution asynchroniously
+        /// Start a new process and wait for its execution to complete
         /// </summary>
-        public static Task<(string stdout, string stderr)> Execute(
-            string fileName, string commandLineArgs, string cwd = null, bool stdout = true, bool stderr = true)
+        public static string Execute(string fileName, string commandLineArgs, string cwd = null, bool stdout = true)
         {
-            Debug.Assert(!string.IsNullOrEmpty(fileName));
-
-            var tcs = new TaskCompletionSource<(string, string)>();
-
-            var error = new StringBuilder();
-            var output = new StringBuilder();
             var psi = new ProcessStartInfo
             {
                 FileName = fileName,
                 WorkingDirectory = cwd,
                 Arguments = commandLineArgs,
                 UseShellExecute = false,
                 RedirectStandardOutput = stdout,
-                RedirectStandardError = stderr,
-            };
-
-            var process = new Process
-            {
-                EnableRaisingEvents = true,
-                StartInfo = psi,
+                RedirectStandardError = false,
             };
 
-            if (stdout)
-            {
-                process.OutputDataReceived += (sender, e) => output.AppendLine(e.Data);
-            }
-            if (stderr)
-            {
-                process.ErrorDataReceived += (sender, e) => error.AppendLine(e.Data);
-            }
-
-            var processExited = new object();
-            process.Exited += (a, b) =>
-            {
-                lock (processExited)
-                {
-                    // Wait for exit here to ensure the standard output/error is flushed.
-                    process.WaitForExit();
-                }
-
-                if (process.ExitCode == 0)
-                {
-                    tcs.TrySetResult((output.ToString().Trim(), error.ToString().Trim()));
-                }
-                else
-                {
-                    var message = $"'\"{fileName}\" {commandLineArgs}' failed in directory '{cwd}' with exit code {process.ExitCode}: \nSTDOUT:'{output}'\nSTDERR:\n'{error}'";
-
-                    tcs.TrySetException(new InvalidOperationException(message));
-                }
-            };
+            var process = Process.Start(psi);
+            var result = stdout ? process.StandardOutput.ReadToEnd() : null;
```

Comment:
will this meet out of buffer issue?

Reply:
I think it shouldn't hang as long as `ReadToEnd` is called before `WaitForExit` and we only read `stdout`. Test it with `git log` on azure-docs, it completed successfully.